src/main/java/org/spongepowered/common/mixin/core/world/storage/MapStorageMixin.java

```diff
@@ -35,12 +41,50 @@
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
 import org.spongepowered.common.SpongeImpl;
+import org.spongepowered.common.data.manipulator.mutable.item.SpongeMapItemData;
+import org.spongepowered.common.world.WorldManager;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.Optional;
 
 @Mixin(MapStorage.class)
-public abstract class MapStorageMixin {
+public abstract class MapStorageMixin implements org.spongepowered.api.world.map.MapStorage {
```

Comment:
If this implements just the API, this mixin needs to go to org.spongepowered.common.mixin.api

If this has injections in it, you need to separate out the API implementation into the above namespace. If the API needs something set via an injection, the API calls a bridge method added via a bridge.

https://github.com/SpongePowered/SpongeCommon/wiki/Implementation-Policy:-Bridges-and-Mixins